### PR TITLE
Use precompiled headers

### PR DESCRIPTION
**Pull request by sbrodehl**
See #1186 .

### CODE REVIEW

**Comment by sbrodehl** [comment]
Here are the runtimes (in minutes):
### appveyor
- **VS14** 31 min *(0.79x , prev. 39 min)*
- **VS15** 27 min *(0.72x , prev. 37 min)*

### travis
- **python** 45 min *(0.81x , prev. 55min)*
- **python3** 46 min *(0.83x , prev. 55 min)*
- **debug** 33 min *(0.84x , prev. 39 min)*

**Comment by zoq** [comment]
Really nice improvement.

File: src/mlpack/CMakeLists.txt [hunk 1]
```diff
@@ -58,6 +57,15 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   endif()
 endif()
 
+set_target_properties(mlpack PROPERTIES COTIRE_CXX_PREFIX_HEADER_INIT "prereqs.hpp")
+cotire(mlpack)
+
+get_target_property(_mlpackPrefixHeader mlpack COTIRE_CXX_PREFIX_HEADER)
```
Comment (rcurtin) [inline src/mlpack/CMakeLists.txt]: Is this needed?  I don't think `_mlpackPrefixHeader` is used anywhere (maybe I missed it?).

Comment (sbrodehl) [reply src/mlpack/CMakeLists.txt]: No it's not, tried using it for the bindings / cli executables.

**Comment by sbrodehl** [comment]
We can merge this for now, I gonna look into a few other things in the next weeks.

**Comment by rcurtin** [comment]
Thanks again!  This is a nice improvement.

**Pull request merged**